addons/mail/static/tests/helpers/mock_server.js

```diff
@@ -292,6 +292,15 @@ MockServer.include({
         }
         return this._super(route, args);
     },
+    /**
+     * @override
+     */
+    _writeRecord: function (model, values, id, params) {
```

Comment:
Strange to override write, there is no such thing on server code?

Reply:
In the server, tracked values are created when the `write` method is called. On the other hand, during tests, records are given to the mock server. I think the closest thing we could do would be to override the write method and create automatically the tracking values if the model has the `tracking_value_ids` field. But this wouldn't be so good because this would result in a lot of useless records created during tests.. We could remove this override and specify which field we should write on (old_value_XXX, new_value_XXX) but once again, it would be a lot of changes and this would add complexity to the tests.. That's why I just kept the old keys (old_value, new_value) and the explicit tracking values creation. 

What do you think ?

Reply:
"Useless" records are not a problem, it's good actually, if these exist in the server they should do in the tests too.

Not all field changes are tracked, only those that are marked as such. And we typically never change those fields in our tests, and when we do, we would expect to see the tracking change probably (currently each tracking test has to manually setup the right tracking value, it would be much easier if that was built in).

Reply:
Ok, I'm working on it :)